### PR TITLE
Switch to fly.io & Cloudflare

### LINKED ISSUE #913: Switch to EC2
This is not a priority but I think fargate really [struggles on IO / CPU](https://filia-aleks.medium.com/ec2-versus-fargate-performance-comparison-34b1002fbbaa) due to the mixed instances problem and [25% CPU time](https://github.com/aws/containers-roadmap/issues/163).

We can get just fine on t4g nanos for a while with 1 container per instance. Even these alone are cheaper than the smallest fargate task :/

### PR DESCRIPTION
**Pull request by joswayski**
Closes #913

### CODE REVIEW

File: packages/infra/utils/config.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { InstanceClass, InstanceSize, InstanceType } from "aws-cdk-lib/aws-ec2";
+
+export const MIN_NUMBER_OF_INSTANCES = 1;
```
Comment (joswayski) [inline packages/infra/utils/config.ts:3]: TODO unused

File: packages/infra/utils/createEc2Service.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    443, // HTTPS from tasks
+    27017 // MongoDB from tasks - TODO Replace with Private Link in the future
+  ];
+
+  outboundPorts.forEach((port) => {
+    natGatewayProvider.connections.allowFrom(autoScalingGroup, Port.tcp(port));
+  });
+
+  return ec2Service;
+  //   const fargateService = new ApplicationLoadBalancedFargateService(
```
Comment (joswayski) [inline packages/infra/utils/createEc2Service.ts:137]: TODO add scaling back

File: packages/api/src/controllers/health/get.ts [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
-      message: "Saul Goodman",
-      db
-    });
-    // TODO we should log the DB error separately
+    res.status(200).json({ message: "Saul Goodman" });
+    console.log(`Health check successful at ${new Date().toISOString()}`);
   } catch (error) {
-    res.status(500).json({ message: "Error connecting to MongoDB!", error });
+    console.error(error);
+    res.status(500).json({ message: "Error on teh health check!!", error });
```
Comment (ScreamingTaco) [inline packages/api/src/controllers/health/get.ts:10]: Typo

**Comment by joswayski** [comment]
nvm i hate this so much
image

**Pull request merged**